File: pdu/python/sensor_msgs/pdu_cdr_conv_Temperature.py
from .pdu_pytype_Temperature import Temperature
from ..pdu_cdr_runtime import CdrReader, CdrWriter

from ..std_msgs.pdu_cdr_conv_Header import *
from ..builtin_interfaces.pdu_cdr_conv_Time import *



def py_to_cdr_body_Temperature(writer: CdrWriter, src: Temperature):
    py_to_cdr_body_Header(writer, src.header)
    writer.write_float64(src.temperature)
    writer.write_float64(src.variance)


def cdr_body_to_py_Temperature(reader: CdrReader, dst: Temperature):
    cdr_body_to_py_Header(reader, dst.header)
    dst.temperature = reader.read_float64()
    dst.variance = reader.read_float64()
    return dst


def py_to_cdr_Temperature(src: Temperature) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_Temperature(writer, src)
    return writer.bytes()


def cdr_to_py_Temperature(cdr_payload) -> Temperature:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = Temperature()
    return cdr_body_to_py_Temperature(reader, dst)

File: pdu/javascript/sensor_msgs/pdu_cdr_conv_Temperature.js
import { Temperature } from './pdu_jstype_Temperature.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';
import { Header } from '../std_msgs/pdu_jstype_Header.js';
import { PduHeaderConverter } from '../std_msgs/pdu_cdr_conv_Header.js';
import { Time } from '../builtin_interfaces/pdu_jstype_Time.js';
import { PduTimeConverter } from '../builtin_interfaces/pdu_cdr_conv_Time.js';


export class PduTemperatureConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { Temperature } src
     */
    static to_cdr_body(writer, src) {
        PduHeaderConverter.to_cdr_body(writer, src.header);
        writer.write_float64(src.temperature);
        writer.write_float64(src.variance);
    }

    /**
     * @param {PduCdrReader} reader
     * @param { Temperature } dst
     * @returns { Temperature }
     */
    static cdr_body_to_js(reader, dst) {
        PduHeaderConverter.cdr_body_to_js(reader, dst.header);
        dst.temperature = reader.read_float64();
        dst.variance = reader.read_float64();
        return dst;
    }

    /**
     * @param { Temperature } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { Temperature }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new Temperature());
    }
}
